skutil/h2o/java_templating/templater.py

```diff
@@ -0,0 +1,64 @@
+from jinja2 import Template, Environment, FileSystemLoader
+from sklearn.externals import six
+from abc import ABCMeta
+import os
+
+
+class Templater(six.with_metaclass(ABCMeta, object)):
```

Comment:
I think due to the nature of the variety of responses I think brainstorming and generation of variants is necessary in order to establish a sound basis for the template(s). There is a notion of template inheritance in jinja which we could leverage. I think this element of the implementation will require more forethought to avoid technical debt and preemptively tackle unforeseen deviations from previous templates.

🎩 
:bowtie: